docs/custom-slashing/dd.template.md

```diff
@@ -0,0 +1,83 @@
+- [x] The "offence" that triggers the slash MAY be specified on L1.
+- [ ] The amount to be slashed MAY be configurable without deploying a new contract.
+- [ ] The threshold of number of validators (M/N) that need to signal/vote for the CustomSlashFactory payload MAY be configurable without deploying a new contract or a governance action.
+
+## Overview
+
+Rename the existing `SlashFactory` contract to `EpochSlashFactory` to denote that it slashes all validators in an epoch. Make a new `CustomSlashFactory` contract which creates a payload to slash a provided list of validators for an explicit offence. The amounts to be slashed for each offence will be provided in the constructor of the `CustomSlashFactory` contract. There will only be one explicit offence right now: "missing attestations". Creating the payload via the `CustomSlashFactory` will emit an event with the payload address, and the validator addresses.
+
+Aztec nodes will listen for these events, and then check if the validator is bad committed the alleged offence. If so, they will vote/signal for the payload on L1.
+
+## Details
+
+The SlasherClient will remain the interface that the SequencerPublisher uses to adjust the transaction it sends to the forwarder contract.
+
+Internally, though, the SlasherClient will monitor two conditions:
+
+- an epoch was not proven, so slash all validators via the `EpochSlashFactory` contract (i.e. the current state, rename the L1 contract from `SlashFactory`)
+- a validator has missed X% (e.g. 90%) of attestations according to the Sentinel, so slash that validator via the `CustomSlashFactory` contract (this is new)
+
+Validators will need to have a way to order the various slashing events they observe.
+
+Their first priority is to slash payloads from the `EpochSlashFactory`, sorted oldest to newest.
+
+Their second priority is to slash payloads from the `CustomSlashFactory` for the "missing attestations" offence, sorted oldest to newest, but folding together payloads until one is found that they disagree with.
+
+They will vote to slash any validator that has missed Y% (e.g. 50%) of attestations according to the Sentinel; there are two percentages to be configured:
+
+- the percentage of attestations missed required to create a payload
+- the percentage of attestations missed required to signal/vote for the payload
+
+Node operators will configure their node to listen to the specific CustomSlashFactory contract and specify the percentage of attestations missed required to create a payload.
+
+This will be done via CLI arguments or environment variables.
+
+## Notes
+
+The amount to slash should be high for testnet (e.g. the minimum stake amount). We can use a different amount for mainnet.
+
+The threshold of number of validators (M/N) that need to signal/vote for the CustomSlashFactory payload will be the same as the number of validators that need to signal/vote for the EpochSlashFactory payload.
```

Comment:
Just making clear, that it is not the factories that are specifying N and M, it is the slashing proposer.